cmd/hub/main.go

```diff
@@ -17,52 +17,33 @@ var clients *cmd.Clients
 
 func init() {
 	cobra.OnInitialize(cmd.InitConfig(hub.Config()))
-	cobra.OnInitialize(cmd.InitConfig(buck.Config()))
 	hub.Init(rootCmd)
 }
 
 func main() {
-	if err := rootCmd.Execute(); err != nil {
-		cmd.Fatal(err)
-	}
+	cmd.ErrCheck(rootCmd.Execute())
 }
 
 var rootCmd = &cobra.Command{
 	Use:   hub.Name,
 	Short: "Hub Client",
 	Long:  `The Hub Client.`,
 	PersistentPreRun: func(c *cobra.Command, args []string) {
-		hub.Config().Viper.SetConfigType("yaml")
-		buck.Config().Viper.SetConfigType("yaml")
-
 		cmd.ExpandConfigVars(hub.Config().Viper, hub.Config().Flags)
 
 		if hub.Config().Viper.GetString("session") == "" && c.Use != "init" && c.Use != "login" {
 			msg := "unauthorized! run `%s` or use `%s` to authorize"
 			cmd.Fatal(errors.New(msg), aurora.Cyan(hub.Name+" init|login"), aurora.Cyan("--session"))
 		}
 
-		clients = cmd.NewClients(hub.Config().Viper.GetString("api"), true, &ctx{})
+		clients = cmd.NewClients(hub.Config().Viper.GetString("api"), true)
 		hub.SetClients(clients)
-		buck.SetClients(clients)
+		config := local.DefaultConfConfig()
+		config.EnvPrefix = fmt.Sprintf("%s_%s", strings.ToUpper(hub.Name), strings.ToUpper(buck.Name))
+		buck.SetBucks(local.NewBucketsWithAuth(clients, config, hub.Auth))
```

Comment:
`hub` can create an instance of buckets that leverages an auth function, which adds additional context info (session and org)